geoparquet/index.qmd

```diff
@@ -0,0 +1,86 @@
+---
+title: GeoParquet
+subtitle: Guidelines for GeoParquet
+---
+
+# GeoParquet
+
+GeoParquet is an encoding for how to store geospatial vector data (point, lines, polygons) in [Apache Parquet](https://parquet.apache.org/), a popular columnar storage format for tabular data.
+
+Parquet has a wide ecosystem of tools and support; GeoParquet builds on this existing success by defining how to store geometries in the Parquet format. Because GeoParquet is not a separate format, any program that can read Parquet is able to load GeoParquet as well, even if it can't make sense of the geometry information. This is very similar to how GeoTIFF layers geospatial information on top of the existing TIFF image standard.
+
+The two main things that GeoParquet defines on top of Parquet are how to encode geometries in the geometry column and how to include metadata like the geometries' CRS.
+
+GeoParquet is a relatively young format, and the specification has not yet reached a 1.0 release (as of August 2023, it's at [1.0.0-rc.1](https://github.com/opengeospatial/geoparquet/releases/tag/v1.0.0-rc.1)). However, reading and writing GeoParquet has been [supported in GDAL since version 3.5](https://gdal.org/drivers/vector/parquet.html), and thus can be used in programs like GeoPandas and QGIS.
```

Comment:
Small note that GeoPandas 0.13.2's [`read_parquet`](https://geopandas.org/en/v0.13.2/docs/reference/api/geopandas.read_parquet.html) function doesn't actually go through GDAL but `pyarrow`. Using [`read_file(..., engine="pyogrio")`](https://geopandas.org/en/v0.13.2/docs/reference/api/geopandas.read_file.html) would go through GDAL though, but is apparently much slower than through pyarrow, see benchmarks at https://github.com/geopandas/geopandas/issues/2429#issuecomment-1126077276.

Reply:
That should be a callout box, particularly in comparison to other formats (all other ogr formats we recommend pyogrio pretty much all the time right now)

Reply:
> Small note that GeoPandas 0.13.2's [`read_parquet`](https://geopandas.org/en/v0.13.2/docs/reference/api/geopandas.read_parquet.html) function doesn't actually go through GDAL but `pyarrow`.

Yes indeed, and this is important to call out.

> Using [`read_file(..., engine="pyogrio")`](https://geopandas.org/en/v0.13.2/docs/reference/api/geopandas.read_file.html) would go through GDAL though, but is apparently much slower than through pyarrow, see benchmarks at [geopandas/geopandas#2429 (comment)](https://github.com/geopandas/geopandas/issues/2429#issuecomment-1126077276).

The reason this is slower is because the default (original) pyogrio implementation calls GDAL's row-based `GetNextFeature` API, and then on the pyogrio side (in Cython) assembles columnar arrays. For most source datasets that are row based, whether the loop is in GDAL itself or in Cython doesn't matter. But here it does matter! Because effectively GDAL is transposing from column-oriented to row-oriented, and then pyogrio is doing the reverse!

If you use pyogrio's [Arrow interface](https://pyogrio.readthedocs.io/en/latest/api.html#pyogrio.raw.read_arrow), which implements [GDAL's RFC 86](https://gdal.org/development/rfc/rfc86_column_oriented_api.html) then it would likely be much faster, because there's no transpose back and forth. _But_ this is likely still significantly slower than using `pyarrow` directly because GDAL's RFC 86 implementation uses a fixed row group size (currently set to 65536 by default). So if this row group size doesn't match the underlying row group size in the Parquet files, GDAL will have to slice the original row group into the new size.

All of this is kinda moot anyways because the GDAL binary that is built as part of the pyogrio wheels don't include `libarrow`. So unless you build pyogrio from source, trying to load a GeoParquet file will give you "unknown data format".

Reply:
Thoughts on whether I should put the above explanation anywhere? It's probably more technical than we want to put into these docs